### PR TITLE
Starting work on Image

### PR DESCRIPTION
**Pull request by ctacke**

### CODE REVIEW

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics/MicroGraphics/Samples/ImageLoadSample/MeadowApp.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                spiBus: spiBus,
+                chipSelectPin: Device.Pins.D15,
+                dcPin: Device.Pins.D11,
+                resetPin: Device.Pins.D14,
+                width: 240, height: 240, displayColorMode: ColorType.Format16bppRgb565)
+            {
+                IgnoreOutOfBoundsPixels = true
+            };
+
+            Meadow.Foundation.Graphics.
```
Comment (pekkah) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics/MicroGraphics/Samples/ImageLoadSample/MeadowApp.cs:48]: Is there something missing here?

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics/MicroGraphics/Driver/Graphics.MicroGraphics/Buffers/BufferRgb8888.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        {
+            var index = ((y * Width) + x) * 4;
+
+            //split into R,G,B & invert
+            byte r = Buffer[index];
+            byte g = Buffer[index + 1];
+            byte b = Buffer[index + 2];
+            byte a = Buffer[index + 3];
+
+            return new Color(r, g, b);
```
Comment (pekkah) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics/MicroGraphics/Driver/Graphics.MicroGraphics/Buffers/BufferRgb8888.cs:33]: Alpha is lost?

File: Source/Meadow.Foundation.Core/Enums/ColorType.cs [hunk 3]
```diff
@@ -14,6 +14,7 @@ public enum ColorType
         Format16bppRgb555, //not currently used
         Format16bppRgb565, //TFT in 16 bit mode
         Format18bppRgb666, //TFT in 18 bit mode
-        Format24bppRgb888  //24 bit color
+        Format24bppRgb888,  //24 bit color
+        Format32bppRgba  //32 bit color
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Core/Enums/ColorType.cs:18]: minor preference to make this Format32BppRgba8888

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics/MicroGraphics/Driver/Graphics.MicroGraphics/Buffers/BufferRgb8888.cs (hunk 2 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics/MicroGraphics/Driver/Graphics.MicroGraphics/Buffers/BufferRgb8888.cs:33]: +1 we should use the constructor that takes an alpha value - new Color(r, ,g, b, a) .... great catch @pekkah  - I would have missed that :)

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics/MicroGraphics/Driver/Graphics.MicroGraphics/Image.cs [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+using System.IO;
+using System.Linq;
+
+namespace Meadow.Foundation.Graphics
+{
+    public partial class Image
+    {
+        public IDisplayBuffer DisplayBuffer { get; private set; }
+
+        public int Width { get; private set; }
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics/MicroGraphics/Driver/Graphics.MicroGraphics/Image.cs:12]: we could have the buffer own the Width, Height, etc. vs storing a separate copy

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics/MicroGraphics/Driver/Graphics.MicroGraphics/Image.cs [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+                    break;
+                case 1:
+                    DisplayBuffer = new Buffer1(Width, Height, pixelData);
+                    break;
+                default:
+                    throw new NotSupportedException("Unsupported color depth");
+            }
+        }
+
+        private void LoadJpeg(Stream stream)
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics/MicroGraphics/Driver/Graphics.MicroGraphics/Image.cs:150]: Can't wait to use the hardware jpeg decoding

**Pull request merged**